File: src/com/dafttech/terra/graphics/gui/elements/GUIElement.java
package com.dafttech.terra.graphics.gui.elements;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.dafttech.terra.game.Events;
import com.dafttech.terra.game.world.Vector2;
import com.dafttech.terra.graphics.AbstractScreen;
import com.dafttech.terra.graphics.gui.GUIObject;

public abstract class GUIElement extends GUIObject {
    public TextureRegion image;
    
    public GUIElement(Vector2 p, Vector2 s) {
        super(p, s);
    }

    public void draw(AbstractScreen screen, Vector2 origin) {
        screen.batch.begin();
        screen.batch.draw(image, position.x + origin.x, position.y + origin.y);
        screen.batch.end();
        screen.batch.setColor(Color.WHITE);
    }
}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/elements/GUIElement.scala
package com.dafttech.terra.engine.gui.elements

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.dafttech.terra.engine.AbstractScreen
import com.dafttech.terra.engine.gui.GUIObject
import com.dafttech.terra.engine.vector.Vector2d

abstract class GUIElement(p: Vector2d, s: Vector2d) extends GUIObject(p, s) {
  var image: TextureRegion = null

  def draw(screen: AbstractScreen): Unit = {
    val p: Vector2d = getScreenPosition
    screen.batch.begin()
    if (image != null)
      screen.batch.draw(image, p.x.toFloat, p.y.toFloat)
    else
      println(s"img null in $getClass base GUIElement")
    screen.batch.end()
    screen.batch.setColor(Color.WHITE)
  }
}